src/test/java/eu/yals/test/ui/VaadinTest.java

```diff
@@ -0,0 +1,184 @@
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.ProtocolHandshake;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Service class for all Vaadin aka UI Tests
+ *
+ * @since 2.7
+ */
+@Slf4j
+public abstract class VaadinTest extends ParallelTest {
+  static {
+    Logger sauceLabsLog = Logger.getLogger(SauceLabsIntegration.class.getName());
+    Logger selenideRemoteLog = Logger.getLogger(ProtocolHandshake.class.getName());
+
+    sauceLabsLog.setLevel(Level.SEVERE);
+    selenideRemoteLog.setLevel(Level.WARNING);
+  }
+
+  private static final int SERVER_PORT =
+      Integer.parseInt(
+          System.getProperty(TestApp.Properties.SERVER_PORT, TestApp.Defaults.SERVER_PORT));
+  private static final String LOCAL_URL =
+      String.format("http://host.testcontainers.internal:%d", SERVER_PORT);
+  private static final String REPORT_DIRECTORY =
+      System.getProperty(TestApp.Properties.REPORT_DIR, TestApp.Defaults.REPORT_DIR);
+  protected static final String BASE_URL =
+      System.getProperty(TestApp.Properties.TEST_URL, LOCAL_URL);
+
+  private static final String BUILD_NAME =
+      System.getProperty(TestApp.Properties.BUILD_NAME, TestApp.Defaults.BUILD_NAME);
+
+  private static String testName;
+
+  @Rule
+  public final TestRule watchman =
+      new TestWatcher() {
+        @Override
+        protected void starting(Description description) {
+          super.starting(description);
+          testName = setTestNameFromTestDescription(description);
+          log.info(String.format("Starting build '%s'. Test: '%s", BUILD_NAME, testName));
+        }
+
+        @Override
+        protected void succeeded(Description description) {
+          super.succeeded(description);
+          Cookie cookie = new Cookie("zaleniumTestPassed", "true");
+          getDriver().manage().addCookie(cookie);
+        }
+
+        @Override
+        protected void failed(Throwable e, Description description) {
+          Cookie cookie = new Cookie("zaleniumTestPassed", "false");
+          getDriver().manage().addCookie(cookie);
+        }
+      };
+
+  @Before
+  public void setup() throws Exception {
+    addTestNameToDriver();
+
+    super.setup();
+    Parameters.setScreenshotErrorDirectory(REPORT_DIRECTORY);
+  }
+
+  protected void open(String relativeOrAbsoluteUrl) {
+    final String PROTOCOL_MARKER = "://";
+    boolean isUrlAbsolute = relativeOrAbsoluteUrl.contains(PROTOCOL_MARKER);
+    if (isUrlAbsolute) {
+      getDriver().get(relativeOrAbsoluteUrl);
+    } else {
+      getDriver().get(BASE_URL + relativeOrAbsoluteUrl);
+    }
+  }
+
+  protected YalsElement $$(String cssSelector) {
+    return YalsElement.wrap(findElement(By.cssSelector(cssSelector)));
+  }
+
+  protected VaadinElement $$(TestBenchElement element) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The instance method name '$$' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/gh/yadevee/yals/pullRequest?prid=5392475)